test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -374,6 +467,27 @@ void validateResults(ReaderGroupManager readerGroupManager, String readerGroupNa
         readerGroupManager.deleteReaderGroup(readerGroupName);
     }
 
+    void waitForScaling(String scope) throws InterruptedException, ExecutionException {
+        for (int waitCounter = 0; waitCounter < 12; waitCounter++) {
+            StreamSegments streamSegments = controller.getCurrentSegments(scope, AUTO_SCALE_STREAM).get();
```

Comment:
This line is the one which throws InterruptedException and ExecutionException. Use join() or FutureHelper methods to handle it.

Reply:
using join(). removed throws.